### DISCUSSION TITLE
How to trim alpha channel in PNG?

### DISCUSSION THREAD

**Discussion by keto33** [ImageMagick/ImageMagick#2100] [Help] [answered]
If we -trim a PNG and then -flatten, it will reach the original size. It seems the alpha channel dimensions have been preserved.

```
magick z.png -trim z2.png
magick z2.png -flatten z3.png
magick z2.png -background white -flatten z4.png
```

How can we make the -trim to be permanent to make sure that the dimensions will remain the same in any subsequent command?

![z](https://user-images.githubusercontent.com/66027046/83679824-f1b5da00-a5d7-11ea-9f4f-6d6a1fab3a5f.png)
![z2](https://user-images.githubusercontent.com/66027046/83679826-f24e7080-a5d7-11ea-8cb6-81405df815fa.png)
![z3](https://user-images.githubusercontent.com/66027046/83679829-f2e70700-a5d7-11ea-9edc-69aa06a6f950.png)
![z4](https://user-images.githubusercontent.com/66027046/83679831-f37f9d80-a5d7-11ea-8ff3-fd218c68a1cb.png)

**Comment by fmw42** [comment, marked as answer]
PNG keeps track of the original size. Other formats such as JPG do not. To prevent that, add +repage after the -trim.
